modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/UserAccountResourceImpl.java

```diff
@@ -832,14 +846,22 @@ public Page<UserAccount> postAccountUserAccountsByEmailAddress(
 	}
 
 	@Override
-	public void postAccountUserAccountsByExternalReferenceCodeByEmailAddress(
-			String externalReferenceCode, String[] emailAddresses)
+	public Page<UserAccount>
+			postAccountUserAccountsByExternalReferenceCodeByEmailAddress(
+				String externalReferenceCode, String accountRoleIds,
+				String[] emailAddresses)
 		throws Exception {
 
-		for (String emailAddress : emailAddresses) {
-			postAccountUserAccountByExternalReferenceCodeByEmailAddress(
-				externalReferenceCode, emailAddress);
+		AccountEntry accountEntry =
+			_accountEntryService.fetchAccountEntryByExternalReferenceCode(
```

Comment:
can we use the getAccountEntryByExternalReferenceCode also here? So we can avoid having the check of null